index.js

```diff
@@ -161,14 +161,20 @@ class GithubScm extends Scm {
      * @return {Promise}
      */
     _getCheckoutCommand(config) {
-        const checkoutUrl = `https://${config.host}/${config.org}/${config.repo}`;
+        const checkoutUrl = `${config.host}/${config.org}/${config.repo}`;
         const checkoutRef = config.prRef ? config.branch : config.sha; // if PR, use pipeline branch
         const command = [];
+        const checkoutCommand =
+            'if [ -z $SCM_USERNAME ] || [ -z $SCM_ACCESS_TOKEN ]; then ' +
+            `git clone --quiet --progress --branch ${config.branch} ` +
+            `https://${checkoutUrl} $SD_SOURCE_DIR; ` +
+            'else ' +
```

Comment:
I'm not a fan of having this complexity in here.  We really only need one clone command, but something that creates the `$SCM_USERNAME:$SCM_ACCESS_TOKEN@` part and injects it if necessary.